grails-app/services/org/pih/warehouse/data/ProductAssociationDataService.groovy

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.data
+
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.product.ProductAssociation
+import org.pih.warehouse.product.Product
+import org.springframework.validation.BeanPropertyBindingResult
+
+
+class ProductAssociationDataService {
+
+    /**
+     * Validate product association
+     */
+    Boolean validateData(ImportDataCommand command) {
+        log.info "Validate data " + command.filename
+
+        command.data.eachWithIndex { params, index ->
+            ProductAssociation productAssociationInstance = new ProductAssociation(params)
```

Comment:
Now that I have replaced all instances where I used `product.id` with `product.productCode` and my params have 
```
params: [ 
    'product.prductCode': '16661',
    .... 
 ]
```

instead of 
```
params: [ 
    'product.id': '16661',
    .... 
 ]
```
now
` ProductAssociation productAssociationInstance = new ProductAssociation(params)` does not populate `product` and `associatedProduct` fields instead I have to explicitly do
```
productAssociationInstance.product = Product.findByProductCode(params['product.productCode'])
productAssociationInstance.associatedProduct = Product.findByProductCode(params['associatedProduct.productCode'])
```

Reply:
I'd say that's proper behavior. If there is a product code in the file, then you have to find it as you did, because IIRC it has to be an id in order to properly create object from params.